src/main/scala/com/fulcrumgenomics/vcf/filtration/ReadEndSomaticVariantFilter.scala

```diff
@@ -113,37 +117,117 @@ class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
-
-    Map(Info.id -> pMutation)
+    Map(readEndInfoLine.id -> pMutation)
   }
 
   /** Calculates a pair of priors:
-    *   The first is the prior that the genotype is the result of a true somatic mutations
-    *   The second is the prior that the genotype is the result of the end-repair artifact process
+    *
+    *   - `priorMutation` is the prior that the genotype is the result of a true somatic mutation.
+    *   - `priorArtifact` is the prior that the genotype is an artifact resulting from the mechanism specific to the filter
     *
     * Default implementation gives a mutation prior of {{{(maf * 2)^2}}} capped at 0.9999 and an
     * artifact prior of 1 - mutation prior.  This has the effect of creating priors that are heavily
     * skewed towards the artifact at very low MAFs, and towards mutations as the MAF increases towards 0.5.
     */
-  private[filtration] def priors(pileup: Pileup[PileupEntry], maf: Double) : (Double, Double) = {
-    require(maf >= 0 && maf <= 1, s"Maf must be between 0 and 1. Observed maf ${maf} at ${pileup.refName}:${pileup.pos}.")
+  protected def priors(pileup: Pileup[PileupEntry], maf: Double) : (Double, Double) = {
+    require(maf >= 0 && maf <= 1, s"Maf must be between 0 and 1. Observed maf $maf at ${pileup.refName}:${pileup.pos}.")
 
     val m = if (maf != 0) maf else {
       logger.warning(s"No alt allele observations found with selected cutoffs at: ${pileup.refName}:${pileup.pos}")
       1.0 / pileup.depth
     }
-
     val priorMutation = min(pow(m * 2, 2), 0.9999)
     val priorArtifact = 1 - priorMutation
     (priorMutation, priorArtifact)
   }
+}
+
+/**
+  * Filter that examines SNVs to see if they are likely end repair fill-in artifacts.
+  *
+  * End repair fill-in artifacts form during the process of filling in single-stranded 3' overhangs to create a blunt end.
+  * Oxidative stress can damage the single-stranded template and induce the formation of DNA lesions, such as
+  * 8-oxoguanine. These modified bases undergo mismatched pairing, which after PCR appear as mutations at the ends of
+  * reads.
+  *
+  * @param distance The distance from the end of the template that defines the region where this
+  *                 artifact may occur
+  * @param pValueThreshold the pvalue at and below which mutations are filtered as being likely
+  *                        caused by the artifact
+  */
+class EndRepairFillInArtifactLikelihoodFilter(override val distance: Int = 15, override val pValueThreshold: Option[Double] = None)
+  extends ReadEndSomaticVariantFilter {
+
+  /** The VCF header line that describes the INFO key that the read-end filter references. */
+  val readEndInfoLine: VcfInfoHeader = VcfInfoHeader(
+    id          = "ERFAP",
+    count       = VcfCount.Fixed(1),
+    kind        = VcfFieldType.Float,
+    description = s"P-Value for the End Repair Fill-in Artifact Filter test, parameterized with a 5-prime distance of ${distance}bp."
```

Comment:
I changed the names of the single read-end INFO/FILTER header records.

The trait minimally requires a `def` but since we only need a static I drop down to `val` here.

I also parameterized the description for all read-end filters to include the distance parameter.

I realize now I could have also parameterized the p-value threshold too!